packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,179 @@
+ * by **default** will be in effect starting with 3.0. Importantly though, N/N-1 compatibility is still guaranteed with the
+ * proper configurations set.
+ *
+ * TODO: This should be updated to "2.0.0" when 3.0 is released.
+ */
+export const defaultCompatibilityMode = "pre-3.0-default";
+
+/**
+ * Available compression algorithms for op compression.
+ * @legacy
+ * @alpha
+ */
+export enum CompressionAlgorithms {
+	lz4 = "lz4",
+}
+
+/**
+ * @legacy
+ * @alpha
+ */
+export const disabledCompressionConfig: ICompressionRuntimeOptions = {
+	minimumBatchSizeInBytes: Number.POSITIVE_INFINITY,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+export const enabledCompressionConfig = {
+	// Batches with content size exceeding this value will be compressed
+	minimumBatchSizeInBytes: 614400,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+/**
+ * Subset of the IContainerRuntimeOptionsInternal properties which are version-dependent.
+ *
+ * @remarks
+ * When a new option is added to IContainerRuntimeOptionsInternal, we must consider if it's a version-dependent option.
+ * If it's considered version-dependent, then a corresponding entry must be added to `versionDependentOptionConfigs`. If not, then
+ * it must be omitted from this type.
+ *
+ * See [TODO: DOC LINK] for more details.
+ */
+export type IContainerRuntimeOptionsVersionDependent = Required<
+	Omit<
+		IContainerRuntimeOptionsInternal,
+		| "chunkSizeInBytes"
+		| "maxBatchSizeInBytes"
+		| "loadSequenceNumberVerification"
+		| "summaryOptions"
+		| "compatibilityMode"
+	>
+>;
+
+/**
+ * Map of version-dependent IContainerRuntimeOptionsInternal keys to their compatibility related information.
+ * The key is the option name, and the value is an object containing:
+ * - minVersionRequired: The minimum version of the container runtime that is required to use the version-dependent option
+ * - legacyConfig: The default config of the option when the runtime does not meets the min version requirement
+ * - modernConfig: The default config of the option when the runtime meets the min version requirement
+ */
+const versionDependentOptionConfigMap: {
+	[K in keyof IContainerRuntimeOptionsVersionDependent]: {
+		minVersionRequired: string;
+		legacyConfig: IContainerRuntimeOptionsVersionDependent[K];
+		modernConfig: IContainerRuntimeOptionsVersionDependent[K];
+	};
+} = {
+	enableGroupedBatching: {
+		minVersionRequired: "2.0.0",
+		legacyConfig: false,
+		modernConfig: true,
+	},
+	compressionOptions: {
+		minVersionRequired: "2.0.0",
+		legacyConfig: disabledCompressionConfig,
+		modernConfig: enabledCompressionConfig,
+	},
+
+	enableRuntimeIdCompressor: {
+		minVersionRequired: "2.0.0",
+		// For IdCompressorMode `undefined` represents a logical state (off). However, to satisfy the Required<>
+		// constraint we need to have it defined, so we trick the type checker here.
+		legacyConfig: undefined as unknown as "on" | "delayed",
+		modernConfig: "on",
```

Comment:
`enableRuntimeIdCompressor` is an optional feature that host apps can choose to opt into. So here is what I suggest the config be.

`legacyConfig: undefined`
`modernConfig: undefined`